source/beamline/CLS/CLSBeamlineStatus.cpp

```diff
@@ -10,18 +13,20 @@ CLSBeamlineStatus::CLSBeamlineStatus(const QString &name, QObject *parent) :
 	monoMaskControlSet_ = new AMControlSet(this);
 
 	// Setup the basic value options.
-	addOption(On, "On", true);
-	addOption(Off, "Off", true);
+	addOption(On, "Beam is ON", true);
+	addOption(Off, "Beam is OFF", true);
+
+	connect(this, SIGNAL(connected(bool)), this, SLOT(updateBeamlineStatus()));
+	connect(this, SIGNAL(valueChanged(double)), this, SLOT(updateBeamlineStatus()));
+	connect(this, SIGNAL(componentsChanged()), this, SLOT(updateBeamlineStatus()));
 }
 
 CLSBeamlineStatus::~CLSBeamlineStatus()
 {
-	shuttersControlSet_->clear();
-	valvesControlSet_->clear();
-	mirrorMaskControlSet_->clear();
-	monoMaskControlSet_->clear();
+	clearComponents();
```

Comment:
That probably makes sense, as the CLSBeamlineStatus likely isn't the parent of each component.